yaks/lib/yaks/default_policy.rb

```diff
@@ -37,15 +37,21 @@ def derive_mapper_from_object(model)
         CollectionMapper
       else
         klass = model.class
+        splitted_class_name = klass.name.split("::")
+        model_namespace = splitted_class_name[0...-1]
+        model_class_name = splitted_class_name.last
         begin
-          name = klass.name.split('::').last
-          return @options[:namespace].const_get(name + 'Mapper')
+          mapper_class_parts = [*model_namespace, "#{klass.name.split("::").last}Mapper"]
+          return mapper_class_parts.inject(@options[:namespace]) { |prefix, suffix| prefix.const_get(suffix, false) }
```

Comment:
With Jruby and Ruby 1.9.3, we cannot do `A::B.const_get("C::D")`. So we have to do `A::B.const_get("C").const_get("D")`.
`false` here makes sure that we look for superclass mapper before looking for "top level" mappers.